apps/system/js/nfc_handover_manager.js

```diff
@@ -14,689 +13,716 @@
-   */
-  DEBUG: false,
-
-  /**
-   * mozSettings object
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  settings: null,
-
-  /**
-   * mozBluetooth object
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  bluetooth: null,
-
-  /**
-   * mozNfc object
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  nfc: null,
-
-  /**
-   * Default bluetooth adapter
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  defaultAdapter: null,
-
-  /**
-   * Keeps a list of actions that need to be performed after
-   * Bluetooth is turned on.
-   * @type {Array}
-   * @memberof NfcHandoverManager.prototype
-   */
-  actionQueue: [],
-
-  /**
-   * Keeps a list of send file requests made via peer.sendFile(blob).
-   * It will be inspected in the handling of Handover Select messages
-   * to distinguish between static and negotiated handovers.
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  sendFileQueue: [],
-
-  /**
-   * The length of the timeout in milliseconds to wait for an outstanding
-   * handover response.
-   * @type {Number}
-   * @memberof NfcHandoverManager.prototype
-   */
-  responseTimeoutMillis: 9000,
-
-  /**
-   * Set whenever a timeout is defined while waiting for an outstanding handover
-   * response.
-   * @type {Object}
-   * @memberof NfcHandoverManager.prototype
-   */
-  responseTimeoutFunction: null,
-
-  /**
-   * Set to true during a file transfer that was initiated by another device.
-   * @type {boolean}
-   * @memberof NfcHandoverManager.prototype
-   */
-  incomingFileTransferInProgress: false,
-
-  /**
-   * Remembers whether Bluetooth was already saved during an earlier
-   * file transfer.
-   * @type {boolean}
-   * @memberof NfcHandoverManager.prototype
-   */
-  bluetoothStatusSaved: false,
-
-  /**
-   * Remembers whether Bluetooth was enabled or automatically.
-   * @type {boolean}
-   * @memberof NfcHandoverManager.prototype
-   */
-  bluetoothAutoEnabled: false,
-
-  /**
-   * Used to prevent triggering Settings multiple times.
-   * @memberof NfcHandoverManager.prototype
-   */
-  settingsNotified: false,
-
-  /**
-   * Debug method
-   * @param {String} msg debug messages
-   * @param {Object} opObject object to printed after doing JSON.stringfy
-   * @memberof NfcHandoverManager.prototype
-   */
-  _debug: function _debug(msg, optObject) {
-    if (this.DEBUG) {
-      this._logVisibly(msg, optObject);
-    }
-  },
-
-  /**
-   * Logs message in logcat
-   * @param {String} msg debug messages
-   * @param {Object} opObject object to printed after doing JSON.stringfy
-   * @memberof NfcHandoverManager.prototype
-   */
-  _logVisibly: function _logVisibly(msg, optObject) {
-      var output = '[NfcHandoverManager]: ' + msg;
-      if (optObject) {
-        output += JSON.stringify(optObject);
+
+(function() {
+  var NfcHandoverManager = function() {};
```

Comment:
var NfcHandoverManager = function(core) {
  this.core = core;
  this.nfc = core.nfc;
};